_posts/2020-06-07-higher-order-hydras.markdown

```diff
@@ -0,0 +1,491 @@
+Unlike their combinatorial counterparts like TREE and SCG, no search is required to compute these fast-growing function values - just keep applying the transformation rule to the tree until the game says to stop.
+
+Kirby-Paris hydras came first.
+For Kirby-Paris hydras, the rules are simple.
+Start with your hydra, which is an unordered unlabelled rooted tree.
+At each stage, choose a leaf node \\\(C\\\) to chop and a non-negative integer \\\(N\\\).
+If \\\(C\\\) is a child of the root, it is removed from the tree and that's it.
+Otherwise, let \\\(P\\\) be \\\(C\\\)'s parent, and \\\(G\\\) be \\\(P\\\)'s parent.
+Remove \\\(C\\\) from \\\(P\\\), then add \\\(N\\\) copies of the modified \\\(P\\\) as children to \\\(G\\\).
+The game ends when the hydra is reduced to a single node.
+
+To obtain a fast-growing function, we can fix \\\(N\\\), say, \\\(N = 1\\\) at the first step, then \\\(N = 2\\\), \\\(N = 3\\\), and so on,
+and decide on a simple rule for where to cut, say, always choosing the rightmost leaf.
+Then, \\\(Hydra(k)\\\) is the number of steps needed for the game to end starting with a path of length \\\(k\\\), that is, a linear stack of \\\(k + 1\\\) nodes.
+
+Buchholz hydras extend Kirby-Paris hydras with labels, and require a few extra transformation cases.
+\\\(0\\\)-labelled nodes behave the same as in the Kirby-Paris hydra game.
+Integers greater than \\\(0\\\) turn into their predecessor but make the tree deeper.
+\\\(ω\\\), rather than being chopped, turns into the current value of the step counter, more or less.
+I won't explain Buchholz hydras in detail here, but you can read more about them yourself.
+
+Googology wiki has fine articles on
+[Kirby-Paris hydras](https://googology.wikia.org/wiki/Kirby-Paris_hydra) and
+[Buchholz hydras](https://googology.wikia.org/wiki/Buchholz_hydra),
+which I suggest you read before coming back here.
+
+In this post, I introduce a higher extension of these hydras, which I believe is interesting in its own right.
+
+</div>
+
+# An idea to nest hydras
+{: #nested-hydras}
+
+{% include collapser.markdown %}
+
+<div>
+
+In exploring ways to build faster growing functions than the one based on Buchholz hydras, a natural question to ask is "what if we used higher ordinals as labels?"
+and then since hydras are ordinal-like, "what if we used nested hydras as labels?"
+We can skip the details of adapting ordinal transformation rules for hydras right now, but in any case, these nested hydras are surely more powerful than plain Buchholz hydras.
+
+I then thought to add a singleton, which I'll call \\\( \lambda \\\), which when encountered, expands into a hydra involving labels which are hydras involving labels which are hydras... \\\(N\\\) layers deep, containing no mention of \\\( \lambda \\\).
+This is, in some sense, a fixed point ordinal of hydra nesting, or for accuracy, the first common fixed point of whatever ordinal operation nesting hydras corresponds to.
```

Comment:
In my opinion, the comma after "nesting" should be a semi-colon.

Reply:
No, this is grammatically correct. The clauses aren't independent and they are joined already.

Reply:
As a style note, I think it would become less clear if I changed it to use a semicolon. Connecting them here is important.

Reply:
That's also true; re-reading the sentence I think I interpreted the clauses as independent when going over them in my head lol